interfaces/cython/cantera/examples/surface_chemistry/lithium_ion_battery.py

```diff
@@ -0,0 +1,172 @@
+https://doi.org/10.1016/j.electacta.2019.134797
+
+Requires: cantera >= 2.6.0, matplotlib >= 2.0
+Keywords: surface chemistry, kinetics, electrochemistry, battery, plotting
+"""
+
+import cantera as ct
+import numpy as np
+
+# Parameters
+samples = 101
+soc = np.linspace(0., 1., samples)  # [-] Input state of charge (0...1)
+current = -1  # [A] Externally-applied current, negative for discharge
+T = 293  # [K] Temperature
+P = ct.one_atm  # [Pa] Pressure
+
+# Cell properties
+input_file = "lithium_ion_battery.yaml"  # Cantera input file name
+R_electrolyte = 0.0384  # [Ohm] Electrolyte resistance
+area_cathode = 1.1167  # [m^2] Cathode total active material surface area
+area_anode = 0.7824  # [m^2] Anode total active material surface area
+
+# Electrode balancing: The "balancing" of the electrodes relates the chemical
+# composition (lithium mole fraction in the active materials) to the macroscopic
+# cell-level state of charge.
+X_Li_anode_0 = 0.01  # [-] anode Li mole fraction at SOC = 0
+X_Li_anode_1 = 0.75  # [-] anode Li mole fraction at SOC = 100
+X_Li_cathode_0 = 0.99  # [-] cathode Li mole fraction at SOC = 0
+X_Li_cathode_1 = 0.49  # [-] cathode Li mole fraction at SOC = 100
+
+# Calculate mole fractions from SOC
+X_Li_anode = (X_Li_anode_1 - X_Li_anode_0) * soc + X_Li_anode_0
+X_Li_cathode = (X_Li_cathode_0 - X_Li_cathode_1) * (1 - soc) + X_Li_cathode_1
+
+# Import all Cantera phases
+anode = ct.Solution(input_file, "anode")
+cathode = ct.Solution(input_file, "cathode")
+metal = ct.Solution(input_file, "electron")
+electrolyte = ct.Solution(input_file, "electrolyte")
+anode_int = ct.Interface(
+    input_file, "edge_anode_electrolyte", adjacent=[anode, metal, electrolyte])
+cathode_int = ct.Interface(
+    input_file, "edge_cathode_electrolyte", adjacent=[cathode, metal, electrolyte])
+
+# Set the temperatures and pressures of all phases
+for phase in [anode, cathode, metal, electrolyte, anode_int, cathode_int]:
+    phase.TP = T, P
+
+
+# Root finding function
+def newton_solve(f, xstart, C=0.0):
+    """
+    Solve f(x) = C by Newton iteration using initial guess *xstart*
+    """
+    f0 = f(xstart) - C
+    x0 = xstart
+    dx = 1.0e-6
+    n = 0
+    while n < 200:
+        ff = f(x0 + dx) - C
+        dfdx = (ff - f0) / dx
+        step = - f0 / dfdx
+
+        # avoid taking steps too large
+        if abs(step) > 0.1:
+            step = 0.1 * step / abs(step)
+
+        x0 += step
+        emax = 0.00001  # 0.01 mV tolerance
+        if abs(f0) < emax and n > 8:
+            return x0
+        f0 = f(x0) - C
+        n += 1
+    raise Exception("no root!")
+
+
+# This function returns the Cantera calculated anode current (in A)
+def anode_current(phi_s, phi_l, X_Li_anode):
+    """
+    Current from the anode as a function of anode potential relative to
+    electrolyte.
+    """
+    # Set the active material mole fraction
+    anode.X = f"Li[anode]:{X_Li_anode},V[anode]:{1 - X_Li_anode}"
```

Comment:
I know this is borrowed from the Jupyter example, but I think it's better to write this as
```suggestion
    anode.X = {"Li[anode]": X_Li_anode, "V[anode]": 1 - X_Li_anode}
```
(also in cathode_current, and in the test)

Reply:
:tada: ... this was transcribed from MATLAB, but you're absolutely right!